Answered two questions. (#14)
emmawedekind/Front-End-FAQ: README.md

# JavaScript   

**What is a Promise and how do you use it?**  

> A promise in simplest terms is just that. You are calling a function that does dome asyncronous work and and in return it promises to return you a response when it's done. 

Reviewer: ```suggestion
A promise in simplest terms is just that. You are calling a function that does some asynchronous work and and in return it promises to return you a response when it's done. 
```

Reviewer 2: ~Also, there's a second `and`~

Disregard RE https://github.com/emmawedekind/Front-End-FAQ/pull/14#issuecomment-435301616

It is important to understand that you can't attach `.then` to the end of any function. It has to be a function that returns a Promise. A good example of a function that does this is [Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).

You can also make your own functions for asyncronous behaviour that returns a custom Promise. 

```javascript
const getData = new Promise((resolve, reject) => {
    // Do some asyncronous stuff

    // If all goes well
    resolve(/* data to return */);

    // If we had a bad time
    reject(/* Error message and info */)
})
```

If you ever get into a situation where you're nesting promised please first take a look a [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). This is a good way to chain promises.

**Is it fine to learn JavaScript as your first language for web development or is it better to start with HTML and CSS**   

> Definitely learn HTML and CSS first. Many take HTML and mostly CSS understanding for granted. But believe me when I say that without a clear understanding of the DOM and CSS you will really struggle as a Web Developer. I have seen bugs get re-opened over and over and over again for the simplest of CSS mistakes. Not being able to cleanly and effectively assemble your layout structure with a stable set of styles can easily turn a 1-day task into a 2-week nightmare and have you wishing you never picked up this career. 

Reviewer: ```suggestion
Definitely learn HTML and CSS first. Many take HTML and mostly CSS knowledge for granted. Without a clear understanding of the DOM and CSS you will really struggle as a Web Developer. I have seen bugs get re-opened over and over again for the simplest of CSS mistakes. Not being able to cleanly and effectively assemble your layout with a stable set of styles can easily turn a 1-day task into a 2-week nightmare and have you wishing you never picked up this career. 
```

> The big misnomer is that many individuals learn things just enough to be able to Google if they get stuck. CSS isn't one of those things. You can't Google your way out of a CSS bug. Without understanding it, believe me, you can't. The only way is to understand how the cascade works. Once you understand CSS fully then move on to JavaScript. 

Reviewer: ```suggestion
The big misnomer is that many individuals learn things just enough to be able to Google if they get stuck. CSS isn't one of those things. You can't Google your way out of a CSS bug. The only way is to understand how the cascade works. Once you understand CSS fully then move on to JavaScript. 
```
